src/libraries/Common/src/Interop/Browser/Interop.HttpHandlerService.cs

```diff
@@ -0,0 +1,535 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Net.Http;
+using System.IO;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Tasks;
+
+using JSObject = Interop.JavaScript.JSObject;
+using JSException = Interop.JavaScript.JSException;
+using HostObject = Interop.JavaScript.HostObject;
+using Uint8Array = Interop.JavaScript.Uint8Array;
+using Function = Interop.JavaScript.Function;
+
+internal static partial class Interop
+{
+    internal static partial class Browser
+    {
+        internal interface IHttpHandlerService
+        {
+            Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken);
+            void Dispose();
+        }
+
+        public class BrowserHttpHandlerService : IHttpHandlerService, IDisposable
+        {
+
+            private static JSObject? fetch;
+            private static JSObject? window;
+
+            /// <summary>
+            /// Gets whether the current Browser supports streaming responses
+            /// </summary>
+            private static bool StreamingSupported { get; }
+
+            public BrowserHttpHandlerService()
+            { }
+
+            static BrowserHttpHandlerService()
+            {
+                using (var streamingSupported = new Function("return typeof Response !== 'undefined' && 'body' in Response.prototype && typeof ReadableStream === 'function'"))
+                    StreamingSupported = (bool)streamingSupported.Call();
+                handlerInit();
+            }
+
+            private static void handlerInit()
```

Comment:
Why not move them to static ctor and remove ? from the files?

Reply:
handlerInit => HandlerInit (if it's kept... I agree it's not clear why this is factored out)

Reply:
Addressed in recent commit.